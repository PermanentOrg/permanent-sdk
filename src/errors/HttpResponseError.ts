export class HttpResponseError extends Error {
  public readonly statusCode: number;

  public constructor(statusCode: number, message: string) {
    super(message);
    this.statusCode = statusCode;
  }
}
