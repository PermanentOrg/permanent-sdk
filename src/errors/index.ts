export * from './HttpResponseError';
export * from './ValidationError';
