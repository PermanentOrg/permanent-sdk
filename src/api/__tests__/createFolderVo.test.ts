import nock from "nock";
import { createFolderVo } from "..";
import { ValidationError } from "../../errors";

describe("createFolderVo", () => {
	it("should create a folder", async () => {
		nock("https://permanent.local")
			.post("/api/folder/post", {
				displayName: "testFolder",
				parentFolderId: 1,
				failOnDuplicateName: false,
			})
			.replyWithFile(200, `${__dirname}/fixtures/createFolderVo/folder.json`, {
				"Content-Type": "application/json",
			});

		const folderVo = await createFolderVo(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			{
				name: "testFolder",
			},
			{
				id: 1,
			},
			false,
		);

		expect(folderVo).toMatchSnapshot();
	});

	it("should error when provided an invalid response", async () => {
		nock("https://permanent.local")
			.post("/api/folder/post", {
				displayName: "testFolder",
				parentFolderId: 1,
				failOnDuplicateName: false,
			})
			.reply(200, '["Not at all a folderVo"]', {
				"Content-Type": "application/json",
			});

		await expect(
			createFolderVo(
				{
					bearerToken: "12345",
					baseUrl: "https://permanent.local/api",
				},
				{
					name: "testFolder",
				},
				{
					id: 1,
				},
				false,
			),
		).rejects.toBeInstanceOf(ValidationError);
	});
});
