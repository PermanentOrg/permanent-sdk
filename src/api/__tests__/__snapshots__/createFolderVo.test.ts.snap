// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`createFolderVo should create a folder 1`] = `
{
  "AccessVO": null,
  "AccessVOs": null,
  "ChildFolderVOs": [],
  "ChildItemVOs": [],
  "FolderLinkVOs": [],
  "FolderSizeVO": null,
  "LocnVO": null,
  "ParentFolderVOs": [],
  "RecordVOs": [],
  "ShareVOs": [],
  "SharedArchiveVOs": [],
  "TagVOs": [],
  "TimezoneVO": null,
  "accessRole": "access.role.owner",
  "archiveArchiveNbr": null,
  "archiveId": 1,
  "archiveNbr": "0001-0007",
  "createdDT": "2022-12-15T17:04:32",
  "derivedDT": "2022-12-15T17:04:31",
  "derivedEndDT": null,
  "description": null,
  "displayDT": "2022-12-15T17:04:31",
  "displayEndDT": null,
  "displayName": "testFolder",
  "downloadName": "testFolder (2)",
  "downloadNameOk": true,
  "folderId": 7,
  "folder_linkId": 7,
  "folder_linkType": "type.folder_link.private",
  "imageRatio": 1,
  "locnId": null,
  "note": null,
  "parentArchiveNbr": null,
  "parentDisplayName": null,
  "parentFolderId": 3,
  "parentFolder_linkId": 3,
  "pathAsArchiveNbr": [],
  "pathAsFolder_linkId": null,
  "pathAsText": [],
  "posLimit": null,
  "posStart": null,
  "position": 1,
  "publicDT": null,
  "returnDataSize": null,
  "searchScore": null,
  "shareDT": null,
  "sort": "sort.alphabetical_asc",
  "special": null,
  "status": "status.generic.ok",
  "thumbArchiveNbr": null,
  "thumbDT": "2023-06-15T17:04:32",
  "thumbStatus": "status.generic.ok",
  "thumbURL1000": "https://localcdn.permanent.org/_slifty/0001-0007.thumb.w200?t=1686845072&Expires=1686845072&Signature=Mw037PKhP~onRFpdHqN2PZn3uUrARqxlcLFB-QVv3CV3OvMDSurllwtbJVKmb2Ju6pgiN6E3M0u~pYyTAsseaS9H-VQHL65rBHOJEO3deSfNtA9Z9d5soRyQRaGvai1cLR7Ny3x-4nFkP3imj2LASbLBXHq4NrgCzkv18SU-dC82WzIrC-giSfmxuDaJpMYXE5H6wWoCjVOdyUL0v51T7FGavTSLGEEs9PUmCycK6ZGBD~ofLi6T6~jOgs3wkFdK~RSdkncpGD6tffsGxoF7RuGAoGrey8~JCFurI-JJncgOos~yGIdic6YC9LmP3UzuTwH4WjCo8thNDGUSjeyCzw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "thumbURL200": "https://localcdn.permanent.org/_slifty/0001-0007.thumb.w200?t=1686845072&Expires=1686845072&Signature=Mw037PKhP~onRFpdHqN2PZn3uUrARqxlcLFB-QVv3CV3OvMDSurllwtbJVKmb2Ju6pgiN6E3M0u~pYyTAsseaS9H-VQHL65rBHOJEO3deSfNtA9Z9d5soRyQRaGvai1cLR7Ny3x-4nFkP3imj2LASbLBXHq4NrgCzkv18SU-dC82WzIrC-giSfmxuDaJpMYXE5H6wWoCjVOdyUL0v51T7FGavTSLGEEs9PUmCycK6ZGBD~ofLi6T6~jOgs3wkFdK~RSdkncpGD6tffsGxoF7RuGAoGrey8~JCFurI-JJncgOos~yGIdic6YC9LmP3UzuTwH4WjCo8thNDGUSjeyCzw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "thumbURL2000": "https://localcdn.permanent.org/_slifty/0001-0007.thumb.w200?t=1686845072&Expires=1686845072&Signature=Mw037PKhP~onRFpdHqN2PZn3uUrARqxlcLFB-QVv3CV3OvMDSurllwtbJVKmb2Ju6pgiN6E3M0u~pYyTAsseaS9H-VQHL65rBHOJEO3deSfNtA9Z9d5soRyQRaGvai1cLR7Ny3x-4nFkP3imj2LASbLBXHq4NrgCzkv18SU-dC82WzIrC-giSfmxuDaJpMYXE5H6wWoCjVOdyUL0v51T7FGavTSLGEEs9PUmCycK6ZGBD~ofLi6T6~jOgs3wkFdK~RSdkncpGD6tffsGxoF7RuGAoGrey8~JCFurI-JJncgOos~yGIdic6YC9LmP3UzuTwH4WjCo8thNDGUSjeyCzw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "thumbURL500": "https://localcdn.permanent.org/_slifty/0001-0007.thumb.w200?t=1686845072&Expires=1686845072&Signature=Mw037PKhP~onRFpdHqN2PZn3uUrARqxlcLFB-QVv3CV3OvMDSurllwtbJVKmb2Ju6pgiN6E3M0u~pYyTAsseaS9H-VQHL65rBHOJEO3deSfNtA9Z9d5soRyQRaGvai1cLR7Ny3x-4nFkP3imj2LASbLBXHq4NrgCzkv18SU-dC82WzIrC-giSfmxuDaJpMYXE5H6wWoCjVOdyUL0v51T7FGavTSLGEEs9PUmCycK6ZGBD~ofLi6T6~jOgs3wkFdK~RSdkncpGD6tffsGxoF7RuGAoGrey8~JCFurI-JJncgOos~yGIdic6YC9LmP3UzuTwH4WjCo8thNDGUSjeyCzw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "timeZoneId": 88,
  "type": "type.folder.private",
  "updatedDT": "2022-12-15T17:04:32",
  "view": "folder.view.grid",
  "viewProperty": null,
}
`;
